File: maxArea.c
#include<stdio.h>

#define min(a,b) ((a) > (b) ? (b):(a))

int maxArea(int* height, int heightSize){
    int i,j;
    int max=0,now=0;
    for(i=0;i<heightSize-1;i++)
    {
        for(j=i+1;j<heightSize;j++)
        {
            now=min(height[i],height[j])*(j-i);
            if(now > max)
            {
                max=now;
                //printf("%d, %d, %d\n",i,j,max);
            }
        }
        if(height[i] > height[i+1])
            i++;
    }
    return max;
}

int main()
{
    int test[]={1,8,6,2,5,4,8,3,7};
    int out=maxArea(test,9);
    
    printf("%d\n",out);
    return 0;
}
    

File: maxArea.py
from typing import *

class Solution:
    def maxArea(self, height: List[int]) -> int:

        left=0
        right=len(height)-1
        out=min(height[left],height[right])*(right-left)
        while left<right:
            if height[left]<height[right]:
                old=height[left]
                left+=1
                while left<len(height) and old>=height[left]:
                    left+=1
                out=max(out,min(height[left],height[right])*(right-left))
            else:
                old=height[right]
                right-=1
                while right>=0 and old>=height[right]:
                    right-=1
                out=max(out,min(height[left],height[right])*(right-left))
        return out

inp=[1,8,6,2,5,4,8,3,7]
sl=Solution()
print(sl.maxArea(inp))

